src/lib/Service/ContentService.php

```diff
@@ -166,7 +172,7 @@ public function setContent(CoreContent $content, ContentOptions $options): array
         ];
 
         return [
-            'contentId' => $content->id,
+            'contentId' => $this->repositoryConfigResolver->useRemoteId() ? $contentInfo->remoteId : $content->id,
```

Comment:
Okay... So now I understand union type issues earlier. This is not the cleanest architecture. You should have separate fields for ID and RemoteID. Those are not the same by their meaning nor strict type.

Reply:
This is required by reco engine. Content is identified by `contentId` field. Sorry, I forget to mention about this in PR description.